pythonforandroid/bootstraps/webview/build/src/main/java/org/kivy/android/PythonActivity.java

```diff
@@ -181,19 +200,22 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
 
         final Thread wvThread = new Thread(new WebViewLoaderMain(), "WvThread");
         wvThread.start();
+
     }
 
     @Override
     public void onDestroy() {
         Log.i("Destroy", "end of app");
         super.onDestroy();
-        
+
         // make sure all child threads (python_thread) are stopped
         android.os.Process.killProcess(android.os.Process.myPid());
     }
 
     public void loadLibraries() {
-        PythonUtil.loadLibraries(getFilesDir());
+        String app_root = new String(getAppRoot());
+        File app_root_file = new File(app_root);
+        PythonUtil.loadLibraries(app_root_file);
```

Comment:
This, together with the entire pull request https://github.com/kivy/python-for-android/pull/1543/files , is for me definitely the most critical part.

I like the loading mechanism as a general construct since it seems well enough done, however, I don't understand why it is necessary. Shouldn't this all just work through dynamic linking? Because this and the entire code of https://github.com/kivy/python-for-android/pull/1543/files do add quite a lot of additional complexity in the loading process, which to me at a first glance shouldn't be needed, so I'd like to understand what is going on and if we really need & want this